src/pages/workspace/WorkspaceSettingsPage.js

```diff
@@ -39,125 +39,116 @@ const defaultProps = {
     ...policyDefaultProps,
 };
 
-class WorkspaceSettingsPage extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.getCurrencyItems = this.getCurrencyItems.bind(this);
-        this.submit = this.submit.bind(this);
-        this.validate = this.validate.bind(this);
-    }
-
+function WorkspaceSettingsPage(props) {
     /**
      * @returns {Object[]}
      */
-    getCurrencyItems() {
-        const currencyListKeys = _.keys(this.props.currencyList);
+    function getCurrencyItems() {
```

Comment:
Should this be `useCallback` as well? Or is it okay as is?

Reply:
I'd probably reach for `useMemo()` here because we are calling `getCurrencyItems()` and by using `useMemo()` we will avoid having to compute the value again.

Reply:
```js
const currencyItems = useMemo(() => ... 
```

Reply:
Ah nice call!